mypy/semanal.py

```diff
@@ -1600,6 +1594,30 @@ def configure_base_classes(self,
             return
         self.calculate_class_mro(defn, self.object_type)
 
+    def enum_has_final_values(self, base: Instance) -> bool:
+        if (
+            base.type.is_enum
+            and base.type.fullname not in (
+                'enum.Enum', 'enum.IntEnum', 'enum.Flag', 'enum.IntFlag')
+            and base.type.names
+            and base.type.defn
+        ):
+            # TODO: if `Var` ever has `.has_explicit_value` prop, remove `ast` traversal.
+            # Check that enum members are really initialized, ignore methods.
+            # We need at least one assigned member to call this Enum `final`.
+            for node in base.type.defn.defs.body:
```

Comment:
We can't look at the body of a base class, since in incremental mode it may be empty. See `ClassDef.serialize`.

Reply:
Ok, in this case I will try adding `Var.has_explicit_value` 👍 

Thanks!

Reply:
Done, now I am using `Var` and its newly-added `has_explicit_value` flag.